### ISSUE TITLE
Evolving UT^2's battle controller further/again

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#172]
There are two different TWEANN classes in MM-NEAT now. One is the old machinelearning.networks.TWEANN that was used to evolve UT^2. The other is the edu.southwestern.networks.TWEANN class that is used for evolution in MM-NEAT. There are differences, but they are similar enough that it should be possible to once again evolve the battle controller of UT^2 from scratch, the way that I did years ago for BotPrize.

However, I don't want to simply replace the old TWEANN class. I want to maintain full backwards compatibility, and currently, the xml file that is loaded to initialize the battle controller is a TWEANN encoded in the old format. We have to make sure at every step of this process that the ability to load the old saved TWEANN format is never lost. The default configuration of UT^2 should still be loading that particular old TWEANN instance.

So, how do we maintain backwards compatibility while making UT^2 work with the newer TWEANN class? Basically, you need to modify the TWEANNController class which is only used by UT^2. This class will contain two instance variables. One is the old type of TWEANN, and the one you will add will be for the new type of TWEANN (Ask @Squillyprice01 how to deal with two difference classes that have the exact same name). At any time, one of these will be null. Essentially, there will be two ways to initialize the TWEANNController: either with an old TWEANN (this constructor already exists) or a new style of TWEANN (add support for this).

Basically, look everywhere that the currently existing tweann variable is used inside of TWEANNController, and add code that checks if it is null. If that variable is null, then accomplish the job using the new TWEANN (which must not be null), otherwise proceed as normal.

Once again, @Squillyprice01 has a bit of experience with upgrading code in this manner.

**Comment by schrum2** [comment]
It would be cool to do this, but it is now on hold indefinitely.

**Comment by schrum2** [comment]
Sort of unrelated, but note that maintaining backwards compatibility depends on using the wox serialization library forever, but I'm trying to get rid of it #348 . Therefore, I might need to load the archived xml using wox and then serialize/save with whatever new serialization method I eventually get working.
